### ISSUE TITLE
Chrome v 85 break the scroll horizontally in RTL.

### ISSUE THREAD

**Issue by MehdiSaeedifar** [ag-grid/ag-grid#4045]
Chrome v 85 break the scroll horizontally in RTL.

Please check the sample in [this link](https://www.ag-grid.com/javascript-grid-rtl/) to reproduce the bug.

**Comment by gportela85** [comment]
The issue has been patched in the latest branch. Thanks for reporting it

**Comment by MehdiSaeedifar** [comment]
@gportela85 Thanks. Is there any hotfix release? It is break all of our projects in chrome 85.

**Comment by ahmadi3d** [comment]
Same issue, lots of angry users..., can we get a hotfix for this.
Thank you.

**Comment by d3viance** [comment]
Doest someone fix this issue without disabling "CSSOM View Scroll Coordinates"?

And if you do, How?

Thanks.

**Comment by mohsen69** [comment]
Same problem for me ! Please help.

**Comment by gportela85** [comment]
This is the commit that fixed the issue https://github.com/ag-grid/ag-grid/commit/acffa118e0c705d17786377bd43a9fca247cdf72
You can try to manually apply this to your JS files, but a new release should be coming soon.

**Comment by leibole** [comment]
I got the same issue.
When is the new release coming out and how can I use it?

**Comment by majid-n** [comment]
Same problem here. is there any temporary solution until the new release comes ?

**Comment by d3viance** [comment]
> This is the commit that fixed the issue [acffa11](https://github.com/ag-grid/ag-grid/commit/acffa118e0c705d17786377bd43a9fca247cdf72)
> You can try to manually apply this to your JS files, but a new release should be coming soon.

Im using the JavaScript version (using ag-grid-enterprise.min.noStyle.js and ag-grid-community.min.noStyle.js)
How can i use this TypeScript file you attached? I didnt find a way to use it to fix it....

Thank you.

**Comment by mohsen69** [comment]
> This is the commit that fixed the issue [acffa11](https://github.com/ag-grid/ag-grid/commit/acffa118e0c705d17786377bd43a9fca247cdf72)
> You can try to manually apply this to your JS files, but a new release should be coming soon.

I made all these changes but still the problem exists. I'm waiting for new release.

**Comment by MehdiSaeedifar** [comment]
Still waiting for a new release.

**Comment by kudasai20** [comment]
Based off of https://github.com/ag-grid/ag-grid/commit/acffa118e0c705d17786377bd43a9fca247cdf72 I added this to my project and it's working fine.

```
agGrid._.getScrollLeft = function(e, r) { const s = e.scrollLeft; return r ? Math.abs(s) : s; }
agGrid._.setScrollLeft = function(e, v, r) { if (r) { v *= -1; } e.scrollLeft = v; }
```

**Comment by majid-n** [comment]
@kudasai20 where did you add this code and where is **agGrid** imported from ?

**Comment by leibole** [comment]
This is what I added for a react application, in the body of my app.js

```
import { GridPanel } from "ag-grid-community";

if (process.env.NODE_ENV !== "development") {
  LogRocket.init("tqmhgl/sheeping-lido");
}

if (getChromeVersion() === 85) {
  GridPanel.prototype.onBodyHorizontalScroll = function (eSource) {
    var getScrollLeft = function (e, r) {
      const s = e.scrollLeft;
      return r ? Math.abs(s) : s;
    };
    var _a = this.eCenterViewport,
      scrollWidth = _a.scrollWidth,
      clientWidth = _a.clientWidth;

    var scrollLeft = Math.floor(getScrollLeft(eSource, this.enableRtl)); 

    var minScroll = 0;
    var maxScroll = scrollWidth - clientWidth;

    if (scrollLeft < minScroll) {
      scrollLeft = minScroll;
    } else if (scrollLeft > maxScroll) {
      scrollLeft = maxScroll;
    }

    this.doHorizontalScroll(scrollLeft);
    this.resetLastHorizontalScrollElementDebounced();
  };

  GridPanel.prototype.getCenterViewportScrollLeft = function () {
    var getScrollLeft = function (e, r) {
      const s = e.scrollLeft;
      return r ? Math.abs(s) : s;
    };
    return getScrollLeft(this.eCenterViewport, this.enableRtl);
  };
}
```

And thank you! Your comment really helped

**Comment by ahmadi3d** [comment]
@leibole 
I tried your solution but i get the error : 
"this.resetLastHorizontalScrollElementDebounced is not a function"
by the way what did u use for `getChromeVersion` ?

**Comment by leibole** [comment]
```
function getChromeVersion() {
  var raw = navigator.userAgent.match(/Chrom(e|ium)\/([0-9]+)\./);

  return raw ? parseInt(raw[2], 10) : false;
}
```

**Comment by kazazimohamad** [comment]
I fixed it with a custom build.
1 - You can edit dom.ts  file and build community-modules
2 - replace community-modules built folders to grid-packages\ag-grid-community\node_modules\@ag-grid-community
3- build it and enjoy

**Comment by mazensabri** [comment]
Is the last fix released?

**Comment by mohsen69** [comment]
> I fixed it with a custom build.
> 1 - You can edit dom.ts file and build community-modules
> 2 - replace community-modules built folders to grid-packages\ag-grid-community\node_modules@ag-grid-community
> 3- build it and enjoy

@kazazimohamad  What platform do you are using?I'm using it in angular 8 and i can't find **dom.ts** in any directory or sub-directory of **ag-grid-community | ag-grid | ag-grid-angular** folders at the node_modules.

**Comment by IgorLav** [comment]
when we will be able to install the fix?

**Comment by MehdiSaeedifar** [comment]
@mohsen69 @IgorLav It is released via v 24.0.0, and it is available now.

**Comment by meysamsahragard** [comment]
This problem exists in the online demo in v 24.
https://www.ag-grid.com/javascript-grid-rtl/
